### PR TITLE
[GMP] change the data source

### PR DESCRIPTION
**Pull request by fpelliccioni**
**gmp/6.2.1**

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by fpelliccioni** [comment]
Context:
Downloading from original source is failing on Github Actions.
Trying to `wget https://gmplib.org/download/gmp/gmp-6.2.1.tar.bz2` on a GhA workflow:

https://github.com/k-nuth/js-api/actions/runs/5505442508/jobs/10032819990#step:7:13

File: recipes/gmp/all/conandata.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sources:
   "6.2.1":
-    url: "https://gmplib.org/download/gmp/gmp-6.2.1.tar.bz2"
-    sha256: "eae9326beb4158c386e39a356818031bd28f3124cf915f8c5b1dc4c7a36b4d7c"
+    url: "https://github.com/alisw/GMP/archive/refs/tags/v6.2.1.tar.gz"
+    sha256: "fb776e19e0267053c3d6ffef0dcc001fb441da0e2b5224cfd9656aa2f1b8614d"
```
Comment (prince-chrismc) [inline recipes/gmp/all/conandata.yml:4]: I am not a fan of the checksum changing, why are they different?

File: recipes/gmp/all/conandata.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 sources:
   "6.2.1":
-    url: "https://gmplib.org/download/gmp/gmp-6.2.1.tar.bz2"
-    sha256: "eae9326beb4158c386e39a356818031bd28f3124cf915f8c5b1dc4c7a36b4d7c"
+    url: "https://github.com/alisw/GMP/archive/refs/tags/v6.2.1.tar.gz"
```
Comment (prince-chrismc) [inline recipes/gmp/all/conandata.yml:3]: would it be reasonable to use the mirror feature of the URL?

```suggestion
    url: 
      - "https://github.com/alisw/GMP/archive/refs/tags/v6.2.1.tar.gz"
      - "https://gmplib.org/download/gmp/gmp-6.2.1.tar.bz2"
```

File: recipes/gmp/all/conandata.yml (hunk 1 above)
Comment (fpelliccioni) [reply recipes/gmp/all/conandata.yml:4]: Because the original file is `.tar.bz2` and the new one is `.tar.gz`

Comment (fpelliccioni) [reply recipes/gmp/all/conandata.yml:4]: Another option is https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.xz but also will change the checksum

Comment (fpelliccioni) [reply recipes/gmp/all/conandata.yml:4]: https://ftp.gnu.org/gnu/gmp/gmp-6.2.1.tar.bz2 is the right option

**Review by prince-chrismc — APPROVED** [review]
I am not sure whats the best order for the URL mirrors but this makes sesne to me

**Review by ericLemanissier — APPROVED** [review]

**Pull request merged**